services/api/mw2pdf/utils/validation.js

```diff
@@ -0,0 +1,31 @@
+import Joi from 'joi'
+
+export function assertValidPassthroughParameters(passthroughParameters) {
+  Joi.assert(
+    passthroughParameters,
+    Joi.string()
+      .custom(
+        (value, helpers) => {
+          JSON.parse(value)
+          return value
+        },
+        'is a valid JSON string',
+      ),
+  )
+  Joi.assert(
+    JSON.parse(passthroughParameters),
+    Joi.object()
+      .pattern(
+        /\w+/,
```

Comment:
This specifies that we are going to run this _validator_ for every key that matches the /w+/ pattern?

Reply:
Correct -- I'll add a comment explaining that since it is pretty opaque

Reply:
I put it in a named variable instead which I think is more clear.